### ISSUE TITLE
Requested Items can be sold by the merchant NPC

### ISSUE THREAD

**Issue by JeroonBosch** [FakeFishGames/Barotrauma#14375]
### Discussed in https://github.com/FakeFishGames/Barotrauma/discussions/14374

<div type='discussions-op-text'>

<sup>Originally posted by **ubertpendragon** July 31, 2024</sup>
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

Merchants can sell their requested items as they are not exclusive.

This gives very little room to change an outpost's store values because it's possible that you can buy the requested goods then sell them for easy profit.

Requested Items should be removed from the shop list?

### Reproduction steps

.

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Single player

### Version

v1.5.9.1 (Summer Update Hotfix 2)

### Which operating system did you encounter this bug on?

Windows

### Relevant error messages and crash reports

_No response_</div>

**Comment by JeroonBosch** [comment]
> it's not uncommon to create money printing loops
> requested goods and sold goods arent exclusive either

I agree requested goods and sold goods should be exclusive. It makes no sense they can be the same item.

> Merchants can sell their requested items as they are not exclusive.

I think it's okay for them to sell their requested items, but imo they should sell them at a much increased price (they're scarce)... like 5x their normal value.

Some more control over what can appear there is also welcome:
![image](https://github.com/user-attachments/assets/2cd1d144-4935-456e-96a1-98a3351c3c1a)

It would make sense if we could make these thematic: research outposts wanting more alien-related items, military outposts requested lead and stuff, mining outposts wanting explosives for mining.

**Comment by ubertpendragon** [comment]
> I think it's okay for them to sell their requested items, but imo they should sell them at a much increased price (they're scarce)... like 5x their normal value.

Good point that should fix it

Make sure the price modifiers apply to all merchants in the outpost (example: general merchant requests fuel rods, electrical merchant sells fuel rods in the same outpost)


With all the talents and reputation kicking in, I've noticed while trying to mod stores that there's a soft ceiling of around 0.8 `sellpricemodifier` on how high you can get the sell price before you can abuse the buy/sell. You don't want buy price to go below sale price. Realistically it only happens when requested goods are factored

Also, stores are missing a simple `buypricemodifier=""`

**Comment by Jasontti** [comment]
Tested.
Closing as completed
